packages/compiler-sfc/src/rewriteDefault.ts

```diff
@@ -88,8 +65,20 @@ export function rewriteDefault(
   return s.toString()
 }
 
-export function hasDefaultExport(input: string): boolean {
-  return defaultExportRE.test(input) || namedDefaultExportRE.test(input)
+export function hasDefaultExport(ast: Statement[]): boolean {
+  for (const stmt of ast) {
+    if (stmt.type === 'ExportDefaultDeclaration') {
+      return true
+    } else if (
+      stmt.type === 'ExportNamedDeclaration' &&
+      stmt.specifiers.length > 0
+    ) {
+      return stmt.specifiers.some(
```

Comment:
We should return only when default export is found, as multiple named export statements can exist.

Reply:
Fixed.